tests/Api.Integration.Tests/RegisterTests.cs

```diff
@@ -0,0 +1,203 @@
+        var sessionResponse = await registrationBeginResponse.Content.ReadFromJsonAsync<SessionResponse<CredentialCreateOptions>>();
+        sessionResponse.Should().NotBeNull();
+        sessionResponse!.Data.User.DisplayName.Should().Be(tokenRequest.DisplayName);
+        sessionResponse.Data.User.Name.Should().Be(tokenRequest.Username);
+    }
+
+    [Fact]
+    public async Task Client_Can_Complete_Registration_Of_User_Credential_With_Passkey_To_Server()
+    {
+        const string originUrl = "https://bitwarden.com/products/passwordless/";
+        const string rpId = "bitwarden.com";
+
+        var tokenRequest = TokenGenerator.Generate();
+        var tokenResponse = await _client.AddSecretKey().PostAsJsonAsync("/register/token", tokenRequest);
+        var registerTokenResponse = await tokenResponse.Content.ReadFromJsonAsync<RegisterEndpoints.RegisterTokenResponse>();
+
+        var registrationBeginRequest = new FidoRegistrationBeginDTO
+        {
+            Token = registerTokenResponse!.Token,
+            Origin = originUrl,
+            RPID = rpId
+        };
+        var registrationBeginResponse = await _client
+            .AddPublicKey()
+            .PostAsJsonAsync("/register/begin", registrationBeginRequest);
+
+        var sessionResponse = await registrationBeginResponse.Content.ReadFromJsonAsync<SessionResponse<CredentialCreateOptions>>();
+
+        var virtualAuth = new VirtualAuthenticatorOptions()
+            .SetIsUserVerified(true)
+            .SetHasUserVerification(true)
+            .SetIsUserConsenting(true)
+            .SetTransport(VirtualAuthenticatorOptions.Transport.INTERNAL)
+            .SetProtocol(VirtualAuthenticatorOptions.Protocol.CTAP2)
+            .SetHasResidentKey(true);
+
+        var options = new ChromeOptions();
+        options.AddArguments("--no-sandbox", "--disable-dev-shm-usage", "--headless");
+        var driver = new ChromeDriver(options);
+        driver.Url = originUrl;
+        driver.AddVirtualAuthenticator(virtualAuth);
+        var result = driver.ExecuteScript(GetScript(sessionResponse!.Data.ToJson()));
+
+        var resultString = result?.ToString() ?? string.Empty;
+
+        var parsedResult = JsonSerializer.Deserialize<AuthenticatorAttestationRawResponse>(resultString);
+
+        var registerCompleteResponse = await _client.PostAsJsonAsync("/register/complete", new RegistrationCompleteDTO
+        {
+            Origin = originUrl,
+            RPID = rpId,
+            Session = sessionResponse.Session,
+            Response = parsedResult
+        });
+
+        registerCompleteResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+        var registerCompleteToken = await registerCompleteResponse.Content.ReadFromJsonAsync<TokenResponse>();
+        registerCompleteToken!.Token.Should().StartWith("verify_");
+    }
+
+    private static string GetScript(string jsonResponse) => $$"""
+       let registration = {{jsonResponse}};
```

Comment:
Can use language injections here (`lang=javascript`) to get syntax highlighting in Rider. Might not work with interpolated strings though.

```suggestion
    private static string GetScript(string jsonResponse) =>
       // lang=javascript
       $$"""
       let registration = {{jsonResponse}};
```

Stretch goal: maybe refactor this into an embedded resource `.js` file, load it from memory, and provide all inputs through function parameters.